ktxspec.adoc

```diff
@@ -837,7 +897,178 @@ individual `keyAndValueByteLength` fields.
+UInt16 endpointCount;
+UInt16 selectorCount;
+UInt32 endpointsByteLength;
+UInt32 selectorsByteLength;
+UInt32 tablesByteLength;
+UInt32 extendedByteLength;
+
+ImageDesc[imageCount] imageDescs;
+
+Byte[endpointsByteLength] endpointsData
+Byte[selectorsByteLength] selectorsData
+Byte[tablesByteLength] tablesData
+Bytes[extendedByteLength] extendedData
+----
+
+`ImageDesc` is the following structure.
+
+.ImageDesc
+[source,c]
+----
+UInt32 imageFlags
+UInt32 rgbSliceByteOffset
+UInt32 rgbSliceByteLength
+UInt32 alphaSliceByteOffset
+UInt32 alphaSliceByteLength
+----
+
+Descriptions in the `imageDescs` array are in the order layer, face and z_slice as if
+arranged by the following pseudo code.
+[source,c]
+----
+for each level in max(levelCount, 1)
+    for each layer in max (layerCount, 1)
+        for each face in faceCount // 1 or 6
+            for each z_slice in max((pixelDepth of level), 1)
+----
+
+`imageCount` is the total number of images in the Mip Level Array.
+
+[TIP]
+====
+`imageCount` may be calculated as follows:
+[source,c]
+----
+int imageCount = max(levelCount, 1) * max(layerCount, 1) * faceCount * totalPixelDepth;
+
+// where totalPixelDepth can be derived as
+int totalPixelDepth = max(pixelDepth, 1);
+for(int i = 1; i < levelCount; i++)
+    totalPixelDepth += max(pixelDepth >> i, 1);
+----
+====
+
+===== globalFlags
+Flags with information that is the same for all images. The following
+flags are valid:
+[source,c]
+----
+isETC1S = 0x01
+hasAlphaSlices = 0x02
+----
+
+For current Basis Universal, `isETC1S` must be set. The flag is provided
+together with `extendedData` to enable future enhancements such as HDR
+support.
+
+`hasAlphaSlices` indicates the Basis data contains alpha slices.
+
+NOTE: This does not necessarily mean the texture has an alpha component.
+
+===== endpointCount
+The number of endpoints in <<_endpointsdata,endpointsData>>.
+
+===== selectorCount
+The number of selectorss in <<_selectorsdata,selectorsData>>.
+
+===== endpointsByteLength
+The length of <<_endpointsdata,endpointsData>>.
+
+===== selectorsByteLength
+The length of <<_selectorsdata,selectorsData>>.
+
+===== tablesByteLength
+The length of <<_tablesdata,tablesData>>.
+
+===== extendedByteLength
+The length of <<_extendeddata,extendedData>>. Must be 0 if `isETC1S` is set in flags.
+
+===== ImageDesc
+====== imageFlags
+Flags giving information about an individual image. The following flag
+is valid:
+[source,c]
+----
+isIFrame = 0x02
+----
+
+Basis Universal supports inter-frame (video) encoding for 2D slices.
+If `isIFrame` is set the image (frame) is an I frame. That is, it
+does not refer to the previous image. If unset, the image is a P
+frame. If the file is not a _video_ file it must be unset.
+
+_Video_ files have a <<_layercount,`layerCount`>> > 0 and must have
+KTXanimation metadata. See <<KTXanimation>>. The value of
+<<_layercount,`layerCount`>> is the number of frames in the video,
+i.e. layers become the temporal axis.
```

Comment:
Since `KTXanimation` metadata could be used without supercompression, should the definition of _video_ file be put outside of BasisU section?

Reply:
I thought about this when writing it. I agree but could not decide on a good place to put it. How about in section 4 "General Comments"?

Reply:
Seems fine. Probably right after the table with texture types.

Reply:
Will do.